src/test/java/de/tum/in/www1/artemis/util/DatabaseUtilService.java

```diff
@@ -1952,6 +1952,78 @@ public Course addCourseWithOneFinishedTextExercise() {
         return course;
     }
 
+    public Course addCourseWithOneFinishedTextExerciseAndSimilarSubmissions(String similarSubmissionText, int studentsAmount) {
```

Comment:
Do we need this in UtilService? This method and the one below seem highly specific to the plagiarism test. I would recommend putting it there. But that's not critical.

Reply:
This method calls mutliple times an abstract class to generate models and therefore it must be in the utilclass

Reply:
Yes, we need it, as it uses the ModelFactory to generate instances. It's only available from UtilService and not from the test.